jastrow.py

```diff
@@ -201,63 +258,100 @@ def updateinternals(self,e,epos,mask=None):
         self._configscurrent[mask,e,:]=epos[mask,:]
-        self._bvalues[mask,:]+=self._get_deltab(e,epos)[mask,:]
 
 
     def value(self): 
         """  """
-        u=np.einsum("ij,j->i",self._bvalues,self.parameters['bcoeff'])+\
-          np.sum(self._avalues*self.parameters['acoeff'], axis=(2,1))
+        u=np.sum(self._bvalues*self.parameters['bcoeff'], axis=(2,1)) +\
+          np.sum(self._avalues*self.parameters['acoeff'], axis=(3,2,1))
         return (1,u)       
 
 
     def gradient(self,e,epos):
-        """We compute the gradient for electron e as 
+        """We compute the gradient for electron e as
         :math:`\grad_e \ln \Psi_J = \sum_k c_k \sum_{j > e} \grad_e b_k(r_{ej})  + \sum_{i < e} \grad_e b_k(r_{ie}) `
-        So we need to compute the gradient of the b's for these indices. 
+        So we need to compute the gradient of the b's for these indices.
         Note that we need to compute distances between electron position given and the current electron distances.
         We will need this for laplacian() as well"""
         nconf=epos.shape[0]
         ne=self._configscurrent.shape[1]
+        nup = self._mol.nelec[0]
         dnew=eedist_i(self._configscurrent,epos)
 
+        dinew=eidist_i(self._mol.atom_coords(),epos)
+        dinew=dinew.reshape(-1,3)
+
         mask=[True]*ne
         mask[e]=False
         dnew=dnew[:,mask,:]
-        dnew=dnew.reshape(-1,3)
-        
+
         delta=np.zeros((3,nconf))
+        if(e < nup): # Spin up electron selected
+            dnew1= dnew[:,:nup-1,:].reshape(-1,3)
+            dnew2= dnew[:,nup-1:,:].reshape(-1,3)
+            #dnew3= np.zeros((nconf,3))
+        else:        # Spin down electron selected
+            #dnew1= np.zeros((nconf,3))
+            dnew2= dnew[:,:nup,:].reshape(-1,3)
+            dnew3= dnew[:,nup:,].reshape(-1,3)
 
         for c,b in zip(self.parameters['bcoeff'],self.b_basis):
-            delta+=c*np.sum(b.gradient(dnew).reshape(nconf,-1,3),axis=1).T
+            delta+=c[1]*np.sum(b.gradient(dnew2).reshape(nconf,-1,3),axis=1).T
+            if(e < nup):
+                delta+=c[0]*np.sum(b.gradient(dnew1).reshape(nconf,-1,3),axis=1).T
+            else:
+                delta+=c[2]*np.sum(b.gradient(dnew3).reshape(nconf,-1,3),axis=1).T
 
         for c,a in zip(self.parameters['acoeff'],self.a_basis):
-            delta+=c*np.sum(a.gradient(dnew).reshape(nconf,-1,3),axis=1).T
+            delta+=c[int(e>=nup)]*np.sum(a.gradient(dinew).reshape(nconf,-1,3),axis=1).T
 
         return delta
 
 
     def laplacian(self,e,epos):
         """ """
         nconf=epos.shape[0]
+        nup = self._mol.nelec[0]
         ne=self._configscurrent.shape[1]
+        
+        # Get and break up eedist_i
         dnew=eedist_i(self._configscurrent,epos)
         mask=[True]*ne
         mask[e]=False
         dnew=dnew[:,mask,:]
-        dnew=dnew.reshape(-1,3)
+        if(e < nup): # Spin up electron selected
```

Comment:
Same things apply here as above.

Reply:
Precomputed the if statement condition and used that as index